src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,150 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public static class Node<T> {
+        private Node<T> next;
+        private Node<T> prev;
+        private T value;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.next = next;
+            this.prev = prev;
+            this.value = value;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            head = new Node<>(null, value, null);
+            tail = head;
+        } else {
+            Node<T> nodeToAdd = new Node<>(tail, value, null);
+            tail.next = nodeToAdd;
+            tail = nodeToAdd;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIfIndexOutOfBonds(index, size);
```

Comment:
create another separate method to check index
```suggestion
        checkIndexForAddMethod(index);
```